dev/Kozani/KozaniHostRuntime/main.cpp

```diff
@@ -112,60 +115,68 @@ void LogArgs(int argc, PWSTR* argv)
-std::wstring GetRemoteAumidFromManifest()
-{
-    AppInfo appInfo{ AppInfo::Current() };
-    auto appId{ appInfo.Id() };
-
-    auto manifestFile{ appInfo.Package().InstalledLocation().GetFileAsync(c_manifestFileName).get()};
-    XmlDocument xmlDocument{ XmlDocument::LoadFromFileAsync(manifestFile).get() };
-
-    std::wstring xpath{ L"/*[local-name()='Package']/*[local-name()='Applications']/*[local-name()='Application' and @Id='" };
-    xpath += appId.c_str();
-    xpath += L"'][1]";
-
-    LOG_HR_MSG(KOZANI_E_INFO, "appId=%ls, xpath=%ls", appId.c_str(), xpath.c_str());
-
-    auto applicationNode{ xmlDocument.SelectSingleNode(xpath) };
-    THROW_HR_IF_NULL_MSG(HRESULT_FROM_WIN32(ERROR_INVALID_DATA), applicationNode,
-            "Invalid LVP manifest - cannot find Application node with Id=%ls. Manifest xml: %ls",
-            appId.c_str(), xmlDocument.GetXml().c_str());
-
-    auto attributeMap{ applicationNode.Attributes() };
-    auto uap10NS{ winrt::Windows::Foundation::PropertyValue::CreateString(L"http://schemas.microsoft.com/appx/manifest/uap/windows10/10") };
-    auto paramAttributeNode{ attributeMap.GetNamedItemNS(uap10NS, L"Parameters")};
-    THROW_HR_IF_NULL_MSG(HRESULT_FROM_WIN32(ERROR_INVALID_DATA), paramAttributeNode,
-            "Invalid LVP manifest - missing uap10:Parameters attribute in Application element with Id=%ls. Application element xml: %ls",
-            appId.c_str(), applicationNode.GetXml().c_str());
-
-    // The uap10:Parameters attribute value contains command line args with remote app aumid. Example: 
-    // uap10:Parameters="-aumid AbcInc.AbcApp_oiuanmryvpj1y!App"
-    auto commandLineArgs{ winrt::unbox_value<winrt::hstring>(paramAttributeNode.NodeValue()) };
-    LOG_HR_MSG(KOZANI_E_INFO, "commandLineArgs=%ls", commandLineArgs.c_str());
-    return GetRemoteAumidFromCommandLineArgs(commandLineArgs.c_str());
+    auto catalog{ AppExtensions::AppExtensionCatalog::Open(appExtensionName.c_str()) };
+    IVectorView<AppExtensions::AppExtension> appExtensions{ catalog.FindAllAsync().get() };
+    for (const auto appExtension : appExtensions)
+    {
+        if (appExtension.Id() == lvpRelativeId)
+        {
+            const auto properties{ appExtension.GetExtensionPropertiesAsync().get() };
+            const auto property{ properties.Lookup(L"RemoteAppAumid").as<IMap<winrt::hstring, winrt::Windows::Foundation::IInspectable>>() };
+            const auto remoteAumid{ winrt::unbox_value<winrt::hstring>(property.Lookup(L"#text")) };
+
+            auto otherData{ properties.Lookup(L"OtherData").as<IMap<winrt::hstring, winrt::Windows::Foundation::IInspectable>>() };
+            const auto otherDataText{ winrt::unbox_value<winrt::hstring>(otherData.Lookup(L"#text")) };
+
+            auto moreotherData{ properties.Lookup(L"MoreOtherData").as<IMap<winrt::hstring, winrt::Windows::Foundation::IInspectable>>() };
```

Comment:
What's MoreOtherData? I don't see it mentioned in the XML snippet comment above

Reply:
Removed. Used for testing, forgot to remove.